libs-scala/contextualized-logging/src/main/scala/com/digitalasset/logging/ContextualizedLogger.scala

```diff
@@ -34,6 +36,41 @@ object ContextualizedLogger {
   def get(clazz: Class[_]): ContextualizedLogger =
     createFor(clazz.getName.stripSuffix("$"))
 
+  private[logging] class TerminationLogger[T](leveledLogger: LeveledLogger)(implicit
```

Comment:
I would go for the akka stage as a last resort. If the purpose of this logger is to log gRPC related closures/related causes, I would consider doing this as on the ServerSubscriber/ServerCallStreamObserver layer...at least on the firs sight it seems straightforward to add such lifecycle loggin there :thinking: